src/pages/storage/StorageVolumeNameLink.tsx

```diff
@@ -23,21 +25,44 @@ const StorageVolumeNameLink: FC<Props> = ({
 }) => {
   const { project } = useCurrentProject();
   const isExternalLink = !hasVolumeDetailPage(volume);
-  const caption = overrideName ? overrideName : volume.name;
+
+  const isInstance =
+    volume.type === "container" || volume.type === "virtual-machine";
+  const { data: instances = [] } = useInstances(volume.project);
+  const instance =
+    isInstance && instances.find((instance) => instance.name === volume.name);
+
+  const isImage = volume.type === "image";
+  const { data: images = [] } = useImagesInProject(volume.project);
+  const image =
+    isImage && images.find((image) => image.fingerprint === volume.name);
+
+  const isCustom = volume.type === "custom";
+  const displayLink = instance || image || isCustom;
+
+  const caption = overrideName
+    ? displayLink
+      ? overrideName
+      : ""
+    : volume.name;
 
   return (
     <div className={classnames("u-flex", className)}>
       <div
         className={classnames("u-truncate", "volume-name-link")}
-        title={caption}
+        title={volume.name}
```

Comment:
The title should match the truncated contents.
```suggestion
        title={caption}
```